pkg/resourcemonitor/podresourcesscanner.go

```diff
@@ -57,58 +56,49 @@ func NewPodResourcesScanner(namespace string, podResourceClient podresourcesapi.
 }
 
 // isWatchable tells if the the given namespace should be watched.
-func (resMon *PodResourcesScanner) isWatchable(podNamespace string, podName string, hasDevice bool) (bool, bool, error) {
-	pod, err := resMon.k8sClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+// In Scan(), if watchable is false, this pods scan will skip
+// so we can return directly if pod's namespace is not watchable
+func (resMon *PodResourcesScanner) isWatchable(podResource *podresourcesapi.PodResources) (bool, bool, error) {
+	if resMon.namespace != "*" && resMon.namespace != podResource.Namespace {
+		return false, false, nil
+	}
+
+	pod, err := resMon.k8sClient.CoreV1().Pods(podResource.Namespace).Get(context.TODO(), podResource.Name, metav1.GetOptions{})
 	if err != nil {
 		return false, false, err
 	}
 
-	isIntegralGuaranteed := hasExclusiveCPUs(pod)
+	isPodHasIntegralCPUs := podHasIntegralCPUs(pod)
```

Comment:
nit: not a blocker by any means but the name `isPodHas...` is just a funny. Some suggestions for renaming, as the PR needs to be udpated in any case because of build failures

```suggestion
	podHasExclusiveCPUs := checkPodExclusiveCPUs(pod)
```

OR

```suggestion
	podHasExclusiveCPUs := hasExclusiveCPUs(pod)
```